src/modules/musig/musig.md

```diff
@@ -0,0 +1,189 @@
+MuSig - Rogue-Key-Resistant Multisignatures Module
```

Comment:
Should this PR also remove some of the redundant documentation from the include file? In particular the things that @real-or-random mentioned in the original MuSig PR and that we were not explicitly discouraging serializing the state.

Reply:
I think the redundancy is fine. Users will probably read this document zero or one times, vs the include file which is the obvious location of the API reference.

Reply:
Indeed, I will get very confused since I check the include file.

Reply:
I mean mostly the part about the memoryless operation with the counter (https://github.com/ElementsProject/secp256k1-zkp/pull/35#discussion_r251845939). Imo we shouldn't mention this when we don't really have support for it. Sounds a bit confusing for people who *just* read the include file and not not the `.md` because its at such a prominent position. Maybe include file should also mention this (`.md) file?

Reply:
Agreed, we should add a reference to this document in the include file (and maybe we could eliminate some of the memoryless operation stuff there). Will add a commit.

Reply:
Added reference.